drivers/ethernet/Kconfig.sam_gmac

```diff
@@ -22,7 +22,12 @@ config ETH_SAM_GMAC_NAME
 config ETH_SAM_GMAC_QUEUES
 	int "Number of hardware TX and RX queues"
 	default 1
-	range 1 3
+	range 1 1 if SOC_SERIES_SAM4E
+	range 1 3 if !SOC_SERIES_SAM4E && \
+		     !SOC_ATMEL_SAME70_REVB && \
+		     !SOC_ATMEL_SAMV71_REVB
+	range 1 6 if SOC_ATMEL_SAME70_REVB || \
+		     SOC_ATMEL_SAMV71_REVB
```

Comment:
This seems like the kinda thing we should encode in the device tree.

Reply:
@galak Agreed.

@nandojve On second thought, since not initialising all priority queues create DMA problems and this is therefore not really "configurable", device tree would be a better place for this to go in.

Reply:
> @nandojve On second thought, since not initialising all priority queues create DMA problems and this is therefore not really "configurable", device tree would be a better place for this to go in.

This configure the number of HW queues that app will use. The intention is save memory for small applications and it's a SW configuration.

The DMA problem was fixed by right definition on  **drivers/ethernet/eth_sam_gmac_priv.h** by @dgloeck. On top of it, I complemented to make sure devices with 1 queue will work too. Now, system initialize all queues always but properly select the active and inactive ones.

My intention was not create anything new, just fix! I'm open to do changes here if necessary.

Reply:
> This seems like the kinda thing we should encode in the device tree.

I'm not complete aware about DT power on Zephyr but my first thought was: we need diferentiate variations A from B, we will duplicate all DT definitions to add this entry, that entry will not be used for any code generation and we need that information on Kconfig. The user will not be allowed to change it and this will be a long path just to drop 3 range definitions that already uses symbols from Kconfig.

This is a WIP, so if we need to change it, we will do. In this particular case, I'll try to make sure we are on the right path because things are a little _gray_ for me if we need use DT here.

What is the best in this case, keep it or move to DT?

Reply:
> The DMA problem was fixed by right definition on **drivers/ethernet/eth_sam_gmac_priv.h** by @dgloeck. On top of it, I complemented to make sure devices with 1 queue will work too. Now, system initialize all queues always but properly select the active and inactive ones.

In that case, this is indeed configurable and should stay as a Kconfig symbol.

Ideally, the device tree should specify the (maximum) number of queues supported in the hardware (i.e. `GMAC_QUEUE_NUM`), and the `ETH_SAM_GMAC_QUEUES` Kconfig symbol should specify the number of queues activated/used.

Noting that there is currently no GMAC definition in the dts for some reason, I suppose we could keep this Kconfig symbol with the emphasis that it refers to the "number of **active** hardware TX and RX queues," and add GMAC DT binding/instances later in a separate PR.

Reply:
Sounds perfect! May I suggest create a enhanced. So, we can step back think the best way to do it. Maybe can it be done for 2.3? So, we fix now but get compromisse to solve for next release. You can assign it to me. We don't need to be in rush let's just focus on bug fix.

What do you think?

Reply:
@nandojve I have opened the issue #22997 for adding the GMAC device tree definition.

I agree that we should focus on the bugfix for now and try to get this PR in for 2.2.